src/main/scala/mesosphere/marathon/api/serialization/ContainerSerializer.scala

```diff
@@ -53,12 +53,18 @@ object ContainerSerializer {
     container match {
       case _: Container.Mesos =>
         builder.setType(mesos.Protos.ContainerInfo.Type.MESOS)
-      case docker: Container.Docker =>
+      case dd: Container.DockerDocker =>
         builder.setType(mesos.Protos.ContainerInfo.Type.DOCKER)
-        builder.setDocker(DockerSerializer.toMesos(docker))
+        builder.setDocker(DockerDockerSerializer.toMesos(dd))
+      case md: Container.MesosDocker =>
+        builder.setType(mesos.Protos.ContainerInfo.Type.MESOS)
+        builder.setMesos(MesosDockerSerializer.toMesos(md))
+      case ma: Container.MesosAppC =>
+        builder.setType(mesos.Protos.ContainerInfo.Type.MESOS)
+        builder.setMesos(MesosAppCSerializer.toMesos(ma))
     }
 
-    container.volumes.foreach {
+    container.volumes foreach {
```

Comment:
We don't have a preference towards dotless syntax, so I suggest to revert this change and keep the changed lines small.